root/programs/MultiPurposeAuthSite/MultiPurposeAuthSite/Controllers/AccountController.cs

```diff
@@ -0,0 +1,2221 @@
+                                                     new AccountAgreementViewModel
+                                                     {
+                                                         UserId = user.Id,
+                                                         Code = "dummy",
+                                                         Agreement = GetContentOfLetter.Get("Agreement", CustomEncode.UTF_8, null),
+                                                         AcceptedAgreement = false
+                                                     });
+                                            }
+                                            else
+                                            {
+                                                // Login画面へ遷移
+                                                return View("Login");
+                                            }
+                                        }
+                                    }
+                                    else
+                                    {
+                                        // 再作成(CreateAsync)に失敗
+                                        AddErrors(result);
+                                        // 再表示
+                                        return View(model);
+                                    }
+                                }
+                                else
+                                {
+                                    // 削除(DeleteAsync)に失敗
+                                    AddErrors(result);
+                                    // 再表示
+                                    return View(model);
+                                }
+                            }
+
+                            #endregion
+                        }
+
+
+                        #endregion
+                    }
+
+                    // UserManager.CreateAsyncの
+                    // resultのエラー情報を追加
+                    AddErrors(result);
+                    // 再表示
+                    return View(model);
+
+                    #endregion
+                }
+                else
+                {
+                    // uidが空文字列の場合。
+                }
+            }
+            else
+            {
+                // AccountRegisterViewModelの検証に失敗
+            }
+
+            // 再表示
+            return View(model);
+        }
+
+        #endregion
+
+        // → 「サインアップ」から遷移
+
+        #region メアド検証
+
+        /// <summary>
+        /// メアド検証画面（メールからのリンクで結果表示）
+        /// GET: /Account/EmailConfirmation
+        /// </summary>
+        /// <param name="userId">string</param>
+        /// <param name="code">string</param>
+        /// <returns>ActionResultを非同期に返す</returns>
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<ActionResult> EmailConfirmation(string userId, string code)
+        {
+            // 入力の検証
+            if (userId == null || code == null)
```

Comment:
@daisukenishino 

Is the check that `userId` or `code` is not empty unnecessary?

```cs
if (string.IsNullOrEmpty(userId) || string.IsNullOrEmpty(code))
```